src/i2c.rs

```diff
@@ -106,6 +115,34 @@ impl embedded_hal_async::i2c::Error for Error {
     }
 }
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum ListenError {
+    InvalidAddressList,
+}
+
+#[derive(Debug, Eq, PartialEq, Hash)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum ListenCommand<'a> {
+    /// Part of a master's write has been received
+    PartialWrite(&'a [u8]),
+    /// The final part of a master's write has been received
+    WriteFinished(&'a [u8]),
```

Comment:
Interface-wise it would be nicer if the WriteFinished no longer has a remaining buffer, but is just a unitary enum variant without arguments. Currently in practice you'll find the code between PartialWrite and WriteFinished handling the buffer to be duplicated by the end user.

This means you'll emit an extra ListenCommand most of the time, but I think it's worth it.

Reply:
Fixed